translib/transformer/sw_portchannel.go

```diff
@@ -0,0 +1,451 @@
+	} else {
+		log.V(3).Info("Minlinks set to 1 (dafault value)")
+		min_links, err := strconv.Atoi(DEFAULT_PORTCHANNEL_MIN_LINKS)
+		if err != nil {
+			errStr := "Conversion of string to int failed"
+			return errors.New(errStr)
+		}
+		links = min_links
+	}
+	lagInfoMap[*ifName].Field["min-links"] = strconv.Itoa(links)
+
+	log.Infof("Updated the lag-info-map for Interface: %s", *ifName)
+
+	return err
+}
+
+// YangToDb_lag_min_links_xfmr is a Yang to DB translation overloaded method for handle min-links config
+var YangToDb_lag_min_links_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+	if log.V(3) {
+		log.Info("Entering YangToDb_lag_min_links_xfmr")
+	}
+	res_map := make(map[string]string)
+	var err error
+
+	pathInfo := NewPathInfo(inParams.uri)
+	ifKey := pathInfo.Var("name")
+
+	log.Infof("Received Min links config for path: %s; template: %s vars: %v ifKey: %s", pathInfo.Path, pathInfo.YangPath, pathInfo.Vars, ifKey)
+
+	if inParams.param == nil {
+		if log.V(3) {
+			log.Info("YangToDb_lag_min_links_xfmr Error: No Params")
+		}
+		return res_map, err
+	}
+
+	minLinks, _ := inParams.param.(*uint16)
+
+	if int(*minLinks) > 32 || int(*minLinks) < 0 {
+		errStr := "Min links value is invalid for the PortChannel: " + ifKey
+		log.Info(errStr)
+		err = tlerr.InvalidArgsError{Format: errStr}
+		return res_map, err
+	}
+
+	res_map["min_links"] = strconv.Itoa(int(*minLinks))
+	return res_map, nil
+}
+
+var DbToYang_lag_min_links_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	if log.V(3) {
+		log.Info("Entering DbToYang_lag_min_links_xfmr")
+	}
+	var err error
+	result := make(map[string]interface{})
+
+	err = validatePortChannel(inParams.d, inParams.key)
+	if err != nil {
+		log.Infof("DbToYang_lag_min_links_xfmr Error: %v ", err)
+		return result, err
+	}
+	data := (*inParams.dbDataMap)[inParams.curDb]
+	links, ok := data[PORTCHANNEL_TABLE][inParams.key].Field["min_links"]
+	if ok {
+		linksUint16, err := uint16Conv(links)
+		if err != nil {
+			return result, err
+		}
+		result["min-links"] = linksUint16
+	} else {
+		if log.V(3) {
+			log.Info("min-links set to 1 (default value)")
+		}
+		linksUint16, err := uint16Conv(DEFAULT_PORTCHANNEL_MIN_LINKS)
+		if err != nil {
+			return result, err
+		}
+		result["min-links"] = linksUint16
+	}
+
+	return result, err
+}
+
+// DbToYang_intf_lag_state_xfmr is a DB to Yang translation overloaded method for PortChannel GET operation
+var DbToYang_intf_lag_state_xfmr SubTreeXfmrDbToYang = func(inParams XfmrParams) error {
+	var err error
+
+	intfsObj := getIntfsRoot(inParams.ygRoot)
+	if intfsObj == nil || intfsObj.Interface == nil {
+		errStr := "Failed to Get root object!"
+		log.Errorf(errStr)
+		return errors.New(errStr)
+	}
+	pathInfo := NewPathInfo(inParams.uri)
+	ifName := pathInfo.Var("name")
+	if _, ok := intfsObj.Interface[ifName]; !ok {
+		obj, _ := intfsObj.NewInterface(ifName)
+		ygot.BuildEmptyTree(obj)
+	}
```

Comment:
This subtree transformer is annotated at "interfaces/interface/aggregate/state", therefore absence of intfObj should be an error condition, instead of created intfsObjNewInterface

Reply:
There is a chance this xfmr gets called w/o memory allocated for intfObj if GET is called for top-level (i.e interfaces container). This ensures that we allocate memory for intfObj and return it's information. Discussed internally and decided we should keep code.